### PR TITLE
Spinup-VpcsbyId

### PR DESCRIPTION
**Pull request by nvnyale**

### CODE REVIEW

File: ec2/filters.go [hunk 1]
```diff
@@ -66,3 +66,12 @@ func isAvailable() *ec2.Filter {
 		),
 	}
 }
+
+func withVPCID(id string) *ec2.Filter {
```
Comment (tenyo) [inline ec2/filters.go:70]: 👀 `withVpcId`

File: api/routes.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func (s *server) routes() {
 	api.HandleFunc("/{account}/images", s.ImageListHandler).Methods(http.MethodGet)
 	api.HandleFunc("/{account}/images/{id}", s.ImageGetHandler).Methods(http.MethodGet)
 	api.HandleFunc("/{account}/vpcs", s.VpcListHandler).Methods(http.MethodGet)
-	api.HandleFunc("/{account}/vpcs/{id}", s.ProxyRequestHandler).Methods(http.MethodGet)
+	api.HandleFunc("/{account}/vpcs/{id}", s.VpcByIdHandler).Methods(http.MethodGet)
```
Comment (tenyo) [inline api/routes.go:58]: 👀 `VpcShowHandler`

File: api/types.go [hunk 3]
```diff
@@ -537,3 +537,33 @@ func toEc2SecurityGroupResponse(sg *ec2.SecurityGroup) *Ec2SecurityGroupResponse
...
 }
+
+type Ec2VpcResponse struct {
+	Id              string            `json:"id"`
+	CIDRBlock       string            `json:"cidr_block"`
+	DHCPOptionsId   string            `json:"dhcp_options_id"`
+	State           string            `json:"state"`
+	InstanceTenancy string            `json:"instance_tenancy"`
+	IsDefault       bool              `json:"is_default"`
+	Tags            map[string]string `json:"tags"`
```
Comment (tenyo) [inline api/types.go:548]: This should be `[]map[string]string` to match the existing format

File: api/types.go [hunk 4]
```diff
@@ -537,3 +537,33 @@ func toEc2SecurityGroupResponse(sg *ec2.SecurityGroup) *Ec2SecurityGroupResponse
...
+		CIDRBlock:       aws.StringValue(vpc.CidrBlock),
+		DHCPOptionsId:   aws.StringValue(vpc.DhcpOptionsId),
+		State:           aws.StringValue(vpc.State),
+		InstanceTenancy: aws.StringValue(vpc.InstanceTenancy),
+		IsDefault:       aws.BoolValue(vpc.IsDefault),
+		Tags:            tagsMap(vpc.Tags),
+	}
+}
+
+func tagsMap(tags []*ec2.Tag) map[string]string {
```
Comment (tenyo) [inline api/types.go:563]: You'll also need to modify this, see above how the others are done.

File: ec2/vpc.go [hunk 5]
```diff
@@ -28,3 +29,18 @@ func (e *Ec2) ListVPCs(ctx context.Context) ([]map[string]string, error) {
...
+	out, err := e.Service.DescribeVpcsWithContext(ctx, &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			withVPCID(id),
+		},
+	})
+	if err != nil {
+		return nil, ErrCode("describing vpc", err)
+	}
+	if len(out.Vpcs) == 0 {
+		return nil, fmt.Errorf("cannot find vpc with id -> %s", id)
```
Comment (tenyo) [inline ec2/vpc.go:43]: should return `ErrNotFound`
```
return nil, apierror.New(apierror.ErrNotFound, "vpc not found", nil)
```

File: ec2/vpc.go [hunk 6]
```diff
@@ -28,3 +29,18 @@ func (e *Ec2) ListVPCs(ctx context.Context) ([]map[string]string, error) {
...
+
+func (e *Ec2) GetVPCsByID(ctx context.Context, id string) (*ec2.Vpc, error) {
+	out, err := e.Service.DescribeVpcsWithContext(ctx, &ec2.DescribeVpcsInput{
+		Filters: []*ec2.Filter{
+			withVPCID(id),
+		},
+	})
+	if err != nil {
+		return nil, ErrCode("describing vpc", err)
+	}
```
Comment (tenyo) [inline ec2/vpc.go:41]: In all these functions we usually have a Debug statement here to log the `out`

**Review by tenyo — COMMENTED** [review]
I've added some comments, also let's write a test for `GetVPCsByID`

File: ec2/filters.go (hunk 1 above)
Comment (nvnyale) [reply ec2/filters.go]: Done.

File: api/routes.go (hunk 2 above)
Comment (nvnyale) [reply api/routes.go]: Renamed to VpcShowHandler.

File: api/types.go (hunk 3 above)
Comment (nvnyale) [reply api/types.go]: Done.

File: api/types.go (hunk 4 above)
Comment (nvnyale) [reply api/types.go]: done.

File: ec2/vpc.go (hunk 5 above)
Comment (nvnyale) [reply ec2/vpc.go]: done

File: ec2/vpc.go (hunk 6 above)
Comment (nvnyale) [reply ec2/vpc.go:39]: done.

**Review by tenyo — APPROVED** [review]
Looks good 👍

**Pull request merged**